lib/mainloop.c

```diff
@@ -317,6 +317,7 @@ main_loop_reload_config_prepare(MainLoop *self, GError **error)
 
   service_management_publish_status("Reloading configuration");
 
+  app_config_stopped();
```

Comment:
At this point threads still can be running.

Reply:
True as well, I remember trying to find the right spot, but at the end I may have inserted it incorrectly.

Reply:
I am almost positive I identified a different spot where this needs to go, but somehow messed up at the end. Thanks a lot for finding this. I've now fixed it.